infrastructure/merklemountainrange/src/merklemountainrange.rs

```diff
@@ -338,10 +359,45 @@ where
         }
         if (new_index <= self.get_last_added_index()) && (height >= 0) {
             // is this a valid peak which needs to be bagged
-            peaks.push(self.mmr[new_index].hash.clone());
+            peaks.push(self.mmr[new_index].clone());
             self.find_bagging_indexes(height, new_index, peaks); // lets go look for more peaks
         }
     }
+
+    /// Mark an object as pruned, if the MMR can remove this safely it will
+    pub fn prune_object_hash(&mut self, hash: ObjectHash) -> Result<(), MerkleMountainRangeError> {
+        let object = self.data.get_mut(&hash);
+        if object.is_none() {
+            return Err(MerkleMountainRangeError::ObjectNotFound);
+        };
+        let object = object.unwrap();
+        object.pruned = true;
+        let sibling_index = if is_node_right(object.vec_index) {
+            object.vec_index - 1
+        } else {
+            object.vec_index + 1
+        };
+        let sibling = self.data.get_mut(&self.mmr[sibling_index]);
+        if sibling.is_none() {
+            return Ok(());
+        };
+        let sibling = sibling.unwrap();
+        if sibling.pruned == true {
```

Comment:
As written, you'll only free up the memory associated with the current pruned object when its sibling is pruned - which could be never

I'm wondering if a better approach would be to make `MerkleNode::object` an `Option<T>` rather than T so that you can drop the object as soon as the node is pruned (you'd add a `prune` method to `MerkeNode` to implement the housekeeping).

Reply:
Good catch. But the `MerkleNode` as it currently is doesnt need to know if its pruned or not. I can just delete it.